src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -199,8 +199,8 @@ private bool IsDescendantOf(SchemaObjectType schemaObjectType)
 
             // We need this redundant check as the internal backing schema will not be a part of the discriminator implementations of its base type.
             if (ObjectSchema.DiscriminatorValue == "Unknown" &&
-                ObjectSchema.Parents?.All.Count == 1 &&
-                ObjectSchema.Parents.All.First().Equals(schemaObjectType.ObjectSchema))
+                ObjectSchema.Parents?.Immediate.Count == 1 &&
+                ObjectSchema.Parents.Immediate.First().Equals(schemaObjectType.ObjectSchema))
```

Comment:
we should not use `All` here because All recursively collect all the parent schemas of myself, therefore even if this is a unknown, inheriting from the base, if the base has a parent schema, here `ObjectSchema.Parents.All` will have two elements